src/charm.py

```diff
@@ -80,24 +79,20 @@ def __init__(self, *args):
 
         self.framework.observe(self.on.upgrade_action, self.on_upgrade_action)
 
+    @status.on_error(ops.WaitingStatus("Waiting on valid ceritifcate data"))
     def api_dependencies_ready(self):
         common_name = kubernetes_snaps.get_public_address()
         ca = self.certificates.ca
         client_cert = self.certificates.client_certs_map.get("system:kube-apiserver")
         server_cert = self.certificates.server_certs_map.get(common_name)
-
-        if not ca or not client_cert or not server_cert:
-            status.add(WaitingStatus("Waiting for certificates"))
-            return False
-
-        if not self.etcd.is_ready:
-            status.add(WaitingStatus("Waiting for etcd"))
-            return False
```

Comment:
this is already asserted in a reconcile loop handler which executes before this one -- so it's not necessary